src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -706,12 +706,17 @@ public override void SetAsyncCallbacks(SNIAsyncCallback receiveCallback, SNIAsyn
         /// <returns>SNI error code</returns>
         public override uint SendAsync(SNIPacket packet, SNIAsyncCallback callback = null)
         {
-            SNIAsyncCallback cb = callback ?? _sendCallback;
-            lock (this)
```

Comment:
Note: This lock seemed unnecessary to me, hence removed.
Just wanted to call out, it's removed cautiously.

Reply:
The lock isn't unnecessary. I dug into it at some point in the past and it's there to prevent overlapping send and receive  operations if you take it out you can observe strange behaviour under load. However, since locks don't work in async calls it isn't of benefit in those situations. In reality you should never see contention on those locks because nothing above this layer should ever try multiple operations (apart from sending out of band packets) and those have locks higher up to guard them. Ultimately they were a way to fix the sync version of the problem you're working on for async i think.

Reply:
This one is on SendAsync method, sync method locks are intact.

Reply:
Good to go then.